proc_master.go

```diff
@@ -105,7 +103,7 @@ func (mp *master) setupSignalling() {
 	mp.restarted = make(chan bool)
 	mp.descriptorsReleased = make(chan bool)
 	//read all master process signals
-	signals := make(chan os.Signal)
+	signals := make(chan os.Signal, 1)
```

Comment:
Why switch to buffered ?

Reply:
You can find the related discussion [here](https://github.com/golang/go/issues/45604)